jsapp/js/account/organization/InviteModal.tsx

```diff
@@ -1,31 +1,103 @@
-import {Group, Modal, Stack, Text, TextInput, ModalProps} from '@mantine/core';
+import {Group, Modal, Stack, Text, TextInput, ModalProps, Loader} from '@mantine/core';
 import ButtonNew from 'jsapp/js/components/common/ButtonNew';
 import {Select} from 'jsapp/js/components/common/Select';
+import {useSendMemberInvite} from './membersInviteQuery';
+import {useState} from 'react';
+import {OrganizationUserRole} from './organizationQuery';
+import userExistence from 'js/users/userExistence.store';
+import {useField} from '@mantine/form';
+import {notify} from 'js/utils';
 
 export default function InviteModal(props: ModalProps) {
+  const inviteQuery = useSendMemberInvite();
+
+  const [role, setRole] = useState<string | null>(null);
+
+  async function handleUsernameOrEmailCheck(value: string) {
+    if (value === '' || value.includes('@')) {
+      return null;
+    }
+
+    const checkResult = await userExistence.checkUsername(value);
+    if (checkResult === false) {
+      return t('This username does not exist. Please try again.');
+    } else {
+      return null;
+    }
+  }
+
+  const userOrEmail = useField({
+    initialValue: '',
+
+    validate: handleUsernameOrEmailCheck,
+    validateOnBlur: true,
+  });
+
+  const handleSendInvite = () => {
+    if (role) {
+      inviteQuery
+        .mutateAsync({
+          invitees: [userOrEmail.getValue()],
+          role: role as OrganizationUserRole,
+        })
+        .catch(() => {
+          notify(t('Failed to send invite'), 'error');
+        });
+    }
+  };
+
+  const isValidated =
+    role !== null &&
+    userOrEmail.isDirty() &&
+    !userOrEmail.isValidating &&
+    !userOrEmail.error;
+
   return (
     <Modal
       opened={props.opened}
       onClose={props.onClose}
-      title={t('Invite memebrs to your team')}
+      title={t('Invite members to your team')}
```

Comment:
```suggestion
      title={t('Invite member to your team')}
```
Since we're only doing one member at a time currently

Reply:
Speaking of which, can we also make the text of the button that opens the modal singular?